var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -92,11 +92,12 @@ class PyNumpy(PythonPackage):
     variant('blas',   default=True, description='Build with BLAS support')
     variant('lapack', default=True, description='Build with LAPACK support')
 
-    depends_on('python@2.7:2.8,3.4:', type=('build', 'link', 'run'), when='@:1.15')
-    depends_on('python@2.7:2.8,3.5:', type=('build', 'link', 'run'), when='@1.16')
-    depends_on('python@3.5:', type=('build', 'link', 'run'), when='@1.17:1.18')
-    depends_on('python@3.6:', type=('build', 'link', 'run'), when='@1.19')
-    depends_on('python@3.7:', type=('build', 'link', 'run'), when='@1.20:1.21.1')
+    depends_on('python@2.7:2.8,3.4:3.6', type=('build', 'link', 'run'), when='@:1.13')
+    depends_on('python@2.7:2.8,3.4:3.8', type=('build', 'link', 'run'), when='@1.14:1.15')
+    depends_on('python@2.7:2.8,3.5:3.9', type=('build', 'link', 'run'), when='@1.16')
+    depends_on('python@3.5:3.9', type=('build', 'link', 'run'), when='@1.17:1.18')
+    depends_on('python@3.6:3.10', type=('build', 'link', 'run'), when='@1.19')
+    depends_on('python@3.7:3.10', type=('build', 'link', 'run'), when='@1.20:1.21.1')
     depends_on('python@3.7:3.10', type=('build', 'link', 'run'), when='@1.21.2:1.21')
```

Comment:
These two can be consolidated now that all versions have an upper bound:
```suggestion
    depends_on('python@3.7:3.10', type=('build', 'link', 'run'), when='@1.20:1.21')
```

Reply:
Ah good catch, I think I skimmed the second line and read the upper bound in `@1.21.2:1.21` as `1.22